### DISCUSSION TITLE
JSON output for member forms

### DISCUSSION THREAD

**Discussion by shbchk** [ExpressionEngine/ExpressionEngine#4179] [Feature Requests]
### Suggest an idea for ExpressionEngine

Channel Form is now able to handle AJAX requests and [return JSON](https://docs.expressionengine.com/latest/channels/channel-form/overview.html#json) (didn't test it yet, though). 

We could use modern approach (without page reloading) for the Edit Profile Data and Edit Avatar forms as well.

**Comment by shbchk** [comment]
And would be great to have inline error handling for member forms also

**Reply by intoeetive** [reply]
That's a separate request, and we already have it available in some form (but not ready to release yet) https://github.com/ExpressionEngine/ExpressionEngine/pull/4093

**Comment by Zignature** [comment]
IMHO all the member forms - login, logout, register, reset password, etc. - could do with a JSON response.

**Comment by intoeetive** [comment]
I believe this now can be achieved using `inline_errors="yes"` parameter https://docs.expressionengine.com/latest/templates/form-validation.html#inline-errors
